lib/charms/observability_libs/v0/kubernetes_compute_resources_patch.py

```diff
@@ -0,0 +1,409 @@
+                    raise ValueError(
+                        "Failed to apply cpu resource limit patch: "
+                        "value must be greater than or equal to 0"
+                    )
+                copy["cpu"] = str(value)
+
+            return copy
+
+        return ResourceRequirements(limits=_conv(res_req.limits), requests=_conv(res_req.requests))  # type: ignore[arg-type]
+
+    def is_ready(self):
+        """Reports if the resource patch has been applied and is in effect.
+
+        Returns:
+            bool: A boolean indicating if the service patch has been applied and is in effect.
+        """
+        client = Client()
+        pod = client.get(Pod, name=self._pod, namespace=self._namespace)
+        podspec = self._get_container(self._container_name, pod.spec.containers)  # type: ignore[attr-defined]
+
+        try:
+            ready = self._conv_res_req(self.resource_reqs) == self._conv_res_req(podspec.resources)
+            patched = self._is_patched(client)
+        except (ValueError, ApiError) as e:
+            msg = f"Failed to apply resource limit patch: {e}"
+            logger.error(msg)
+            self.on.patch_failed.emit(message=msg)
+            return False
+
+        return patched and ready
+
+    @classmethod
+    def _get_container(cls, container_name: str, containers: List[Container]) -> Container:
+        """Find our container from the container list, assuming list is unique by name.
+
+        Typically, *.spec.containers[0] is the charm container, and [1] is the (only) workload.
+
+        Raises:
+            StopIteration, if the user-provided container name does not exist in the list.
+
+        Returns:
+            An instance of :class:`Container` whose name matches the given name.
+        """
+        return next(iter(filter(lambda ctr: ctr.name == container_name, containers)))
+
+    def _is_patched(self, client: Client) -> bool:
+        """Reports if the resource patch has been applied to the StatefulSet.
+
+        Returns:
+            bool: A boolean indicating if the service patch has been applied.
+        """
+        statefulset = client.get(StatefulSet, name=self._app, namespace=self._namespace)
+        podspec_tpl = self._get_container(
+            self._container_name,
+            statefulset.spec.template.spec.containers,  # type: ignore[attr-defined]
+        )
+
+        return self._conv_res_req(podspec_tpl.resources) == self._conv_res_req(self.resource_reqs)
+
+    @property
+    def _app(self) -> str:
+        """Name of the current Juju application.
+
+        Returns:
+            str: A string containing the name of the current Juju application.
+        """
+        return self._charm.app.name
+
+    @property
+    def _pod(self) -> str:
+        """Name of the unit's pod.
+
+        Returns:
+            str: A string containing the name of the current unit's pod.
+        """
+        return "-".join(self._charm.unit.name.rsplit("/", 1))
```

Comment:
```suggestion
        return self._charm.unit.name.replace("/", "-", 1)
```

Reply:
The suggestion is opposite to the original:
```python
>>> "one/two/0".replace("/", "-", 1)
'one-two/0'
>>> "-".join("one/two/0".rsplit("/", 1))
'one/two-0'
```

Reply:
Ah I see the difference. Good catch.